### PR TITLE
Generalize array checking and remove `pd.Index` call in `_get_partitions`

### PR DESCRIPTION
**Pull request by quasiben**
This PR does two things:
1) Generalizes the array checking using is_arraylike as well as the list check 
2) Removes what I _think_ is an unnecessary `pd.Index` call . 

This fixes an issue for dask-cudf where users want to index a Dask Dataframe with a cudf/cupy object:
> df.loc[cudf.Series([0,10,50], dtype=np.int32)]

The `pd.Index` call is quite old (https://github.com/dask/dask/pull/1913) and was originally written to handle indexing with lists.  Locally I ran  `dask/dataframe/tests/test_indexing.py` and all tests still pass as well as verifying manually that lists are still supported

- [x] Closes #[xxxx](https://github.com/rapidsai/cudf/issues/11877)
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

cc @VibhuJawa @rjzamora

### CODE REVIEW

**Comment by VibhuJawa** [comment]
Thanks for working on this @quasiben , I expect to see some speed up for the GNN use cases when this lands. 

CC: @eriknw  for awareness too in related to `get_edge_data` and `get_node_data`  for PropertyGraphs.

File: dask/dataframe/indexing.py [hunk 1]
```diff
@@ -209,7 +210,7 @@ def _loc_element(self, iindexer, cindexer):
         return new_dd_object(graph, name, meta=meta, divisions=[iindexer, iindexer])
 
     def _get_partitions(self, keys):
-        if isinstance(keys, (list, np.ndarray)):
+        if isinstance(keys, (list, np.ndarray)) or is_arraylike(keys):
```
Comment (jrbourbeau) [inline dask/dataframe/indexing.py:213]: The `np.ndarray`-type check should be covered by `is_arraylike`, so we should be able to remove it

```suggestion
        if isinstance(keys, list) or is_arraylike(keys):
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @quasiben! Can we add a test for this?

File: dask/dataframe/indexing.py [hunk 2]
```diff
@@ -192,6 +193,7 @@ def _loc_element(self, iindexer, cindexer):
         name = "loc-%s" % tokenize(iindexer, self.obj)
         part = self._get_partitions(iindexer)
 
+        breakpoint()
```
Comment (jrbourbeau) [inline dask/dataframe/indexing.py:196]:
```suggestion
```

Comment (quasiben) [reply dask/dataframe/indexing.py:196]: gah!

Comment (jrbourbeau) [reply dask/dataframe/indexing.py]: No worries -- see https://github.com/dask/dask/pull/9638 for a neat `pre-commit` hook I ran across recently

**Comment by jrbourbeau** [comment]
My guess is the gpuCI failures are unrelated (saw them here yesterday https://github.com/dask/dask/pull/9635#pullrequestreview-1171197377), though it'd be great if you could confirm

**Comment by quasiben** [comment]
@jrbourbeau , yes, they are unrelated.  I filed https://github.com/dask/dask/issues/9639 to track the failures

**Review by jrbourbeau — APPROVED** [review]
Thanks @quasiben

**Pull request merged**